types/k6/experimental/browser.d.ts

```diff
@@ -1911,108 +1911,108 @@ export interface Locator {
 
     /**
      * Checks if the element is editable.
      * @param options Options to use.
      * @returns `true` if the element is editable, `false` otherwise.
      */
-    isEditable(options?: TimeoutOptions): boolean;
+    isEditable(options?: TimeoutOptions): Promise<boolean>;
 
     /**
      * Checks if the element is `enabled`.
      * @param options Options to use.
      * @returns `true` if the element is enabled, `false` otherwise.
      */
-    isEnabled(options?: TimeoutOptions): boolean;
+    isEnabled(options?: TimeoutOptions): Promise<boolean>;
 
     /**
      * Checks if the element is `disabled`.
      * @param options Options to use.
      * @returns `true` if the element is disabled, `false` otherwise.
      */
-    isDisabled(options?: TimeoutOptions): boolean;
+    isDisabled(options?: TimeoutOptions): Promise<boolean>;
 
     /**
      * Checks if the element is `visible`.
      * @returns `true` if the element is visible, `false` otherwise.
      */
-    isVisible(): boolean;
+    isVisible(): Promise<boolean>;
 
     /**
      * Checks if the element is `hidden`.
      * @returns `true` if the element is hidden, `false` otherwise.
      */
-    isHidden(): boolean;
+    isHidden(): Promise<boolean>;
 
     /**
      * Fill an `input`, `textarea` or `contenteditable` element with the provided value.
      * @param value Value to fill for the `input` or `textarea` element.
      * @param options Options to use.
      */
-    fill(value: string, options?: ElementHandleOptions): void;
+    fill(value: string, options?: ElementHandleOptions): Promise<void>;
 
     /**
      * Focuses the element using locator's selector.
      * @param options Options to use.
      */
-    focus(options?: TimeoutOptions): void;
+    focus(options?: TimeoutOptions): Promise<void>;
 
     /**
      * Returns the element attribute value for the given attribute name.
      * @param name Attribute name to retrieve value for.
      * @param options Options to use.
      * @returns Attribute value.
      */
-    getAttribute(name: string, options?: TimeoutOptions): string | null;
+    getAttribute(name: string, options?: TimeoutOptions): Promise<string | null>;
 
     /**
      * Returns the `element.innerHTML`.
      * @param options Options to use.
      * @returns Element's innerHTML.
      */
-    innerHTML(options?: TimeoutOptions): string;
+    innerHTML(options?: TimeoutOptions): Promise<string>;
 
     /**
      * Returns the `element.innerText`.
      * @param options Options to use.
      * @returns Element's innerText.
      */
-    innerText(options?: TimeoutOptions): string;
+    innerText(options?: TimeoutOptions): Promise<string>;
 
     /**
      * Returns the `element.textContent`.
      * @param options Options to use.
      * @returns Element's textContent.
      */
-    textContent(options?: TimeoutOptions): string;
+    textContent(options?: TimeoutOptions): Promise<string | null>;
```

Comment:
Since we added the `null` to our docs, I decided to do the same here.

Reply:
It'll need to be changed in the codebase since we're returning [an empty string](https://github.com/grafana/xk6-browser/blob/ebd1ce39c85b40d23c07a85d0e94c968d54f31eb/common/locator.go#L394-L407) from a quick look.

Reply:
Added this issue as a task in https://github.com/grafana/xk6-browser/issues/1305.